src/gromo/utils/gpu_tracking.py

```diff
@@ -0,0 +1,80 @@
+from types import TracebackType
+from typing import Optional, Self, Type
+
+import numpy as np
+from codecarbon import OfflineEmissionsTracker
```

Comment:
We could add codecarbon as an optional dependency. Right now, we already have two optional sets in pyproject.toml: test and docs, with dependencies related to unit tests and documentation building. 

Is this a hard dependency or an optional one? I clearly prefer if this dependency is optional, as there is a high risk of bugs or dependencies blocks with hardware-related library. If it is optional, we could add another optional set in project.toml, called `energy` or `carbon`, that includes only codecarbon.

Reply:
Only this specific file uses codecarbon and it is used as a context manager in a script. It is fully optional. So far I have not implemented any tests on gpu_tracking because it is quite simple and thus we avoid issues with the CI. We might choose to change the energy tracking library any time.

Reply:
In that case, I suggest we add an optional set (like docs and test) called `carbon`, that contains only codecarbon for now and that we set up a conditional import for codacarbon here (raising an error if it is not installed).